tf-modules/cumulus-rds-tf/main.tf

```diff
@@ -90,7 +106,7 @@ resource "aws_rds_cluster" "cumulus" {
   tags                            = var.tags
   final_snapshot_identifier       = "${var.cluster_identifier}-final-snapshot"
   snapshot_identifier             = var.snapshot_identifier
-  db_cluster_parameter_group_name = aws_rds_cluster_parameter_group.rds_cluster_group.id
+  db_cluster_parameter_group_name = var.enable_upgrade ? aws_rds_cluster_parameter_group.rds_cluster_group_v13.id : aws_rds_cluster_parameter_group.rds_cluster_group[0].id
```

Comment:
So if `enable_upgrade` is `false` this will use the existing parameter group ID? It looks like it's accessed differently (via an array index)?

Reply:
Right - so - the purpose of this is as follows: 

1) We have a case where the DB was deployed with the prior version of this module.  User pulls in new version, has V11 param group and DB and wants to deploy the database
2) A user wants to deploy this module new with engine set to 11.X, upgrade to false, param group to v11
3) A user wants to redeploy with this module version either with an updated DB or an upgraded DB
4) A user wants to use this module to manage the upgrade

The goal here is to be able to have managed TF parameter groups without manual TF intervention when doing the upgrade and/or deploying new, keeping in mind that this is the last serverless V1 upgrade we're going to be doing and attempting to not break users that have existing terraform state files. 

The `enable_upgrade` flag sets the db cluster to use the "new" pg13 parameter group, and by count removes the original pg11 (or other, if they're doing something custom) param group.     If it's set to false, the new module creates a 'new' PG13 param group that's just unused, and the cluster targets the 'old' cluster.    

In a future release we'll disable/remove the old resource from the TF and make the upgrade flag vestigial.    We're going to be forced to move to serverless V2 soon, so there will be additional work to migrate to that solution where we can hopefully do better than this for updates.  

This module was not originally intended to handle production upgrades, however there's perhaps a clear need to do so at this point.

Reply:
Great. Thanks for the rundown!